### PR TITLE
[GEOS-9479] Add Backward compatibility on new Cascaded WMS settings

### PR DESCRIPTION
**Pull request by imranrajjad**
JIRA : https://osgeo-org.atlassian.net/browse/GEOS-9479
## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [x] Committs changing the REST API, or any configuration object, should check it the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java [hunk 1]
```diff
@@ -2529,4 +2530,27 @@ public String toString(Object obj) {
             return obj.toString();
         }
     }
+
+    /** Converter for WMSLayerInfoImpl class ensure backwards compatibility with old directories */
+    class WMSLayerInfoLegacyConverter extends AbstractCatalogInfoConverter {
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java]: This is the WMSLayerInfo converter, no need to add legacy. Or... next time we add a parameter that needs defaulting, are we going to call it LegacyLegacyConverter?

File: src/web/wms/src/test/java/org/geoserver/wms/web/data/publish/WMSLayerConfigTest.java [hunk 2]
```diff
@@ -262,4 +268,144 @@ private boolean cascadedControlsVisible(WicketTester tester) {
                 && tester.getComponentFromLastRenderedPage("form:panel:metaDataCheckBoxContainer")
                         != null;
     }
+
+    @Test
+    public void testWMSCascadeSettingsLegacyBean() throws Exception {
+        // this test asserts that the wmslayer settings do not break
+        // when user has switched from 2.16.2 and less version
```
Comment (aaime) [inline src/web/wms/src/test/java/org/geoserver/wms/web/data/publish/WMSLayerConfigTest.java]: 2.16.2 is affected?

File: src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java (hunk 1 above)
Comment (imranrajjad) [reply src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java]: will change name to WMSLayerInfo

File: src/web/wms/src/test/java/org/geoserver/wms/web/data/publish/WMSLayerConfigTest.java (hunk 2 above)
Comment (imranrajjad) [reply src/web/wms/src/test/java/org/geoserver/wms/web/data/publish/WMSLayerConfigTest.java]: yes

**Comment by nmco** [comment]
Unrelated failure:
```
[INFO] Application Schema Integration Test ................ FAILURE [02:18 min]
```

Looks like @aaime requested changes have been applied. Merging.

@imranrajjad this is a bug fix, hence should be backported to 2.16.x right away.

**Comment by imranrajjad** [comment]
2.16.x backport
https://github.com/geoserver/geoserver/pull/4024

**Pull request merged**